src/Template/Component.php

```diff
@@ -28,4 +29,9 @@ public function getTypeAsString(): string
     {
         return $this->type->describe(VerbosityLevel::typeOnly());
     }
+
+    public function jsonSerialize(): mixed
+    {
+        return [$this->name, $this->getTypeAsString()];
```

Comment:
I would prefer to use also some keys in arrays 'name' => $this->name, 'type' => $this->getTypeAsString()